jobs/build_ova/ansible/main.yml

```diff
@@ -46,21 +49,111 @@
       tags:
         - before-test
 
+
+    - name: Copy SOL generating script
+      become: yes
+      copy:
+        src: ../../../generate-sol-log.sh
+        dest: /var/log/rackhd/generate-sol-log.sh
+      tags:
+        - before-test
+
+    - name: Copy VNC recording script
+      become: yes
+      copy:
+        src: ../../../vnc_record.sh
+        dest: /var/log/rackhd/vnc_record.sh
+      tags:
+        - before-test
+
+    - name: Start SOL Log collecting
+      become: yes
+      shell: |
+        cd /var/log/rackhd
+        nohup bash generate-sol-log.sh > sol.log &
+      tags:
+        - before-test
+
+
+    - name: get BUILD_ID from ENV
+      set_fact: BUILD_ID="{{ lookup('env','BUILD_ID') }}"
+
+
+    - name: Start VNC Recording
+      become: yes
+      shell: |
+        cd /var/log/rackhd
+        export fname_prefix="vNode"
+        if [ ! -z $BUILD_ID ]; then
+             export fname_prefix=${fname_prefix}_b${BUILD_ID}
+        fi
+        nohup bash vnc_record.sh /var/log/rackhd $fname_prefix > vnc_record.log   &
+      tags:
+        - before-test
+
     - name: Restart rackhd
       become: yes
-      service: 
+      service:
         name: rackhd
         state: restarted
       tags:
         - before-test
-      
+
+
+#######################
+
+
+    - name: Stop VNC Record, sleep 2 sec to ensure FLV finishes the disk I/O before VM destroyed
+      become: yes
+      shell: |
+          kill -9 $(ps -lef|grep flvrec.py|grep -v grep| awk '{print $4}')
```

Comment:
why not pkill ?
see below.
https://stackoverflow.com/questions/37042643/ansible-shell-task-with-pgrep-keeps-failing-works-in-normal-shell